pkg/agent/route/route_linux.go

```diff
@@ -1363,7 +1364,22 @@ func (c *Client) addServiceCIDRRoute(serviceCIDR *net.IPNet) error {
 			return fmt.Errorf("error listing ip routes: %w", err)
 		}
 		for i := 0; i < len(routes); i++ {
-			if routes[i].Gw.Equal(gw) && !routes[i].Dst.IP.Equal(serviceCIDR.IP) && routes[i].Dst.Contains(serviceCIDR.IP) {
+			// Not the routes we are interested in.
+			if !routes[i].Gw.Equal(gw) {
+				continue
+			}
```

Comment:
Do we need this? According to routes, err := c.listIPRoutesOnGW(), the gateway of the routes should be all gateway.

Reply:
This is not a new condition. I don't think `listIPRoutesOnGW` has filtered routes by next hop.